### ISSUE TITLE
🐞[BUG]: Can not inject any service into @State with Angular 11.1

### ISSUE THREAD

**Issue by motabass** [ngxs/store#1720]
### Affected Package
The issue is caused by package @ngxs/store

### Is this a regression?

 Yes, and no, the previous version of  ANGULAR in which this bug was not present was: 11.0.9

### Description
Since updating Angular to 11.1.0 i am getting a runtime-error `Uncaught ReferenceError: <XYZService> is not defined` when Angular tries to inject my service into the @State-class.

## 🔬 Minimal Reproduction

**Note:** Our policy is that issues reported without a reproduction will be closed immediately and then reopened once a reproduction has been provided. Please respect the developers of this project by doing this. We give of our personal time to work on this project and would rather be spending our time fixing or enhancing the library than chasing down badly described or unreproducable issues.
Please delete this note once you have read it.

I really tried to create a reproduction on stackblitz, but do not get the Error

## 🔥 Exception or Error
<pre><code>
Uncaught ReferenceError: MngMessageHandlerService is not defined
    at Module.nVUX (message.state.ts:20)
    at __webpack_require__ (bootstrap:84)
    at Module.AOOQ (mng-messages.module.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module.a6nC (index.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module.7/V8 (mng-file-select.component.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module.OH68 (index.ts:1)
    at __webpack_require__ (bootstrap:84)
</code></pre>

## Environment

<pre><code>
Libs:
- @angular/core version: 11.1.0
- @ngxs/store version: 3.7.1

Browser:
- [ x] Chrome (desktop) version 88

Others:

I am also using nrwl/nx in my workspace

**Comment by koolkav1** [comment]
I am getting a similar error in Angular version 11.1.0. The issue was not there in Angular version 10*.

**Comment by koolkav1** [comment]
Could I please have your previous package list for a working NGXS package on Angular 11?
   ```
 "@angular/animations": "^11.0.0",
 "@angular/cdk": "^11.1.0",
    "@angular/common": "^11.1.0",
    "@angular/compiler": "^11.1.0",
    "@angular/core": "^11.1.0",
    "@angular/flex-layout": "^11.0.0-beta.33",
    "@angular/forms": "^11.1.0",
    "@angular/material": "^11.1.0",
    "@angular/platform-browser": "^11.1.0",
    "@angular/platform-browser-dynamic": "^11.1.0",
    "@angular/router": "^11.1.0",
```

**Comment by arturovt** [comment]
Hey @motabass 

I wasn't able to reproduce it locally. You can create a reproducible example locally (via `ng new`) and push it to some Git repository, stackblitz is not necessary.

**Comment by fredrikredflag** [comment]
It looks like this is only an issue if you have `emitDecoratorMetadata: true` in your `tsconfig.json`. The issue has been reported in `@angular/angular-cli` and seems to be fixed here: https://github.com/angular/angular-cli/pull/19841

If you don't rely on reflection in runtime you should be able to set `emitDecoratorMetadata: false`

Hopefully it will be released soon. (We use reflection in runtime alot)

**Comment by motabass** [comment]
@fredrikredflag  Thanks for your Investigation! 

I can confirm this is fixed in @angular/cli@11.1.1

**Comment by koolkav1** [comment]
@fredrikredflag, we use reflection in Jest testing. Do you know when a fix will be available?

**Comment by motabass** [comment]
> @fredrikredflag, we use reflection in Jest testing. Do you know when a fix will be available?

The fix is already available in Angular CLI 11.1.1

**Comment by koolkav1** [comment]
> > @fredrikredflag, we use reflection in Jest testing. Do you know when a fix will be available?
> 
> The fix is already available in Angular CLI 11.1.1

Okay, I have updated @angular devkit and the cli. it works, thanks.
